Some extension project fail with 'has no file' error message.

**Describe the bug**
Some extension project fail with 'has no file' error message.

There are some module in multi module project which have no jar artifact, uses karaf-feature extension plugin.

**To Reproduce**

Create a project with     

`
<artifactId>osgi-asciidoc-converter-karaf-features</artifactId>
<packaging>feature</packaging>
`

where the parent project contains the sign plugin definition.

`
[ERROR] Failed to execute goal org.simplify4u.plugins:sign-maven-plugin:0.3.1:sign (default) on project osgi-asciidoc-converter-karaf-features: Execution default of goal org.simplify4u.plugins:sign-maven-plugin:0.3.1:sign failed: Artifact: hu.blackbelt:osgi-asciidoc-converter-karaf-features:feature:1.0.0.20211221_210526_f25d548e_feature_JNG_3407_AddNewGPGAndTagAction has no file -> [Help 1]
`

**Expected behavior**

Build have to be success

--- a/src/it/issue-105/invoker.properties
+++ b/src/it/issue-105/invoker.properties
@@ -1,0 +1,22 @@
+#
+# Copyright 2022 Slawomir Jaranowski
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+invoker.goals = install --no-transfer-progress -Dsettings.security=${project.basedir}/src/it/settings-security.xml
+invoker.mavenOpts = ${argLine}
+
+invoker.environmentVariables.SIGN_KEY_ID = AC71B3E31C0C0D38
+invoker.environmentVariables.SIGN_KEY_PASS = {p01Vm25gEoIH/wDlR3hWNhOH5mn4PBEnSFEtH96k1Yw=}
+
+invoker.ordinal = 100

--- a/src/it/issue-105/issue-105-api/pom-test.xml
+++ b/src/it/issue-105/issue-105-api/pom-test.xml
@@ -1,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2022 Slawomir Jaranowski
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>test-group</groupId>
+        <artifactId>issue-105-parent</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom-test.xml</relativePath>
+    </parent>
+
+    <artifactId>issue-105-api</artifactId>
+    <packaging>bundle</packaging>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>

--- a/src/it/issue-105/issue-105-api/src/main/java/test/TestService.java
+++ b/src/it/issue-105/issue-105-api/src/main/java/test/TestService.java
@@ -1,0 +1,22 @@
+/*
+ * Copyright 2022 Slawomir Jaranowski
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package test;
+
+/**
+ * Test service.
+ */
+public interface TestService {
+}

--- a/src/it/issue-105/issue-105-features/pom-test.xml
+++ b/src/it/issue-105/issue-105-features/pom-test.xml
@@ -1,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2022 Slawomir Jaranowski
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>test-group</groupId>
+        <artifactId>issue-105-parent</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom-test.xml</relativePath>
+    </parent>
+
+    <artifactId>issue-105-features</artifactId>
+    <packaging>feature</packaging>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.karaf.tooling</groupId>
+                <artifactId>karaf-maven-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <enableGeneration>true</enableGeneration>
+                    <aggregateFeatures>true</aggregateFeatures>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/src/it/issue-105/issue-105-features/src/main/feature/feature-required-capabilities.xml
+++ b/src/it/issue-105/issue-105-features/src/main/feature/feature-required-capabilities.xml
@@ -1,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2022 Slawomir Jaranowski
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<features name="asciidoc-converter-required-capabilities-${project.version}"
+    xmlns="http://karaf.apache.org/xmlns/features/v1.5.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.5.0 http://karaf.apache.org/xmlns/features/v1.5.0">
+
+    <!-- These services provided by platform -->
+    <feature name="test-required-capabilities">
+        <capability>osgi.extender;osgi.extender=osgi.component;version=1.3.0</capability>
+    </feature>
+
+</features>

--- a/src/it/issue-105/issue-105-features/src/main/feature/feature.xml
+++ b/src/it/issue-105/issue-105-features/src/main/feature/feature.xml
@@ -1,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2022 Slawomir Jaranowski
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<features name="asciidoc-converter-${project.version}"
+    xmlns="http://karaf.apache.org/xmlns/features/v1.5.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.5.0 http://karaf.apache.org/xmlns/features/v1.5.0">
+
+    <feature name="test-feature" version="${project.version}" description="Test services">
+        <feature prerequisite="true">scr</feature>
+        <bundle>mvn:org.slf4j/slf4j-api/1.7.32</bundle>
+    </feature>
+
+</features>

--- a/src/it/issue-105/pom-test.xml
+++ b/src/it/issue-105/pom-test.xml
@@ -1,0 +1,85 @@
+<!--
+  ~ Copyright 2022 Slawomir Jaranowski
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>test-group</groupId>
+    <artifactId>issue-105-parent</artifactId>
+    <packaging>pom</packaging>
+    <version>1.0.0</version>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <modules>
+        <module>issue-105-api/pom-test.xml</module>
+        <module>issue-105-features/pom-test.xml</module>
+    </modules>
+
+    <build>
+
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.felix</groupId>
+                    <artifactId>maven-bundle-plugin</artifactId>
+                    <version>@maven-bundle-plugin.version@</version>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.karaf.tooling</groupId>
+                    <artifactId>karaf-maven-plugin</artifactId>
+                    <version>4.3.3</version>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-install-plugin</artifactId>
+                    <version>2.5.2</version><!-- MINSTALL-151 -->
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-deploy-plugin</artifactId>
+                    <version>@maven-deploy-plugin.version@</version>
+                </plugin>
+
+            </plugins>
+        </pluginManagement>
+
+        <plugins>
+            <plugin>
+                <groupId>org.simplify4u.plugins</groupId>
+                <artifactId>sign-maven-plugin</artifactId>
+                <version>@project.version@</version>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>sign</goal>
+                        </goals>
+                        <configuration>
+                            <keyFile>@project.basedir@/src/test/resources/priv-key.asc</keyFile>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>

--- a/src/it/verify-sign/pom-test.xml
+++ b/src/it/verify-sign/pom-test.xml
@@ -83,6 +83,25 @@
             <classifier>c1</classifier>
         </dependency>
 
+        <dependency>
+            <groupId>test-group</groupId>
+            <artifactId>issue-105-parent</artifactId>
+            <type>pom</type>
+            <version>1.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>test-group</groupId>
+            <artifactId>issue-105-api</artifactId>
+            <version>1.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>test-group</groupId>
+            <artifactId>issue-105-features</artifactId>
+            <version>1.0.0</version>
+            <classifier>features</classifier>
+            <type>xml</type>
+        </dependency>
+
     </dependencies>
 
     <build>

--- a/src/main/java/org/simplify4u/plugins/sign/SignMojo.java
+++ b/src/main/java/org/simplify4u/plugins/sign/SignMojo.java
@@ -208,8 +208,18 @@ public class SignMojo extends AbstractMojo {
         // collect artifact to sign
         Set<Artifact> artifactsToSign = new HashSet<>();
 
-        artifactsToSign.add(new ProjectArtifact(project));
-        artifactsToSign.add(project.getArtifact());
+        ProjectArtifact projectArtifact = new ProjectArtifact(project);
+        artifactsToSign.add(projectArtifact);
+
+        Artifact artifact = project.getArtifact();
+        if (artifact.getFile() != null) {
+            artifactsToSign.add(artifact);
+        } else {
+            if (!artifact.equals(projectArtifact)) {
+                LOGGER.info("Primary artifact doesn't have the file to sign, continue with attached artifacts.");
+            }
+        }
+
         artifactsToSign.addAll(project.getAttachedArtifacts());
 
         // sign and attach signature to project

